launch.go

```diff
@@ -54,27 +54,23 @@ type scmPath struct {
 	Branch string
 }
 
-func (s scmPath) String() string {
-	return fmt.Sprintf("git@%s:%s/%s#%s", s.Host, s.Org, s.Repo, s.Branch)
-}
-
 func (s scmPath) httpsString() string {
 	return fmt.Sprintf("https://%s/%s/%s#%s", s.Host, s.Org, s.Repo, s.Branch)
 }
 
-// e.g. "git@github.com:screwdriver-cd/launch.git#master"
-func parseScmURL(url string) (scmPath, error) {
-	r := regexp.MustCompile("^git@(.*):(.*)/(.*?)(?:\\.git)?#(.*)$")
-	matched := r.FindStringSubmatch(url)
-	if matched == nil || len(matched) != 5 {
-		return scmPath{}, fmt.Errorf("unable to parse SCM URL %v, match: %q", url, matched)
+func parseScmUri(scmUri, scmName, scmBranch string) (scmPath, error) {
```

Comment:
We were worried the `url` would be different for github vs some other scm and thought using the scmUri, name, and branch would be a more reliable choice